detox/test/e2e/15.urls.test.js

```diff
@@ -1,23 +1,46 @@
 describe('Open URLs', () => {
 
-  it('device.launchApp({{newInstance: true, url: url}) should launch app and trigger handling open url handling in app', async () => {
-    const url = 'detoxtesturlscheme://such-string';
-    await device.launchApp({newInstance: true, url: url});
-    await expect(element(by.text(url))).toBeVisible();
+  const withDefaultArgs = () => ({
+    url: 'detoxtesturlscheme://such-string',
+    launchArgs: undefined,
   });
 
-  it('device.openURL({url: url}) should trigger open url handling in app when app is in foreground', async () => {
-    const url = 'detoxtesturlscheme://such-string';
-    await device.launchApp({newInstance: true});
-    await device.openURL({url: url});
-    await expect(element(by.text(url))).toBeVisible();
+  const withAndroidSingleTaskActivityArgs = () => ({
+    url: 'detoxtesturlscheme.singletask://such-string',
+    launchArgs: Object.freeze({ androidSingleTaskActivity: true }),
   });
 
-  it('device.launchApp({url: url}) should trigger open url handling in app when app is in background', async () => {
-    const url = 'detoxtesturlscheme://such-string';
-    await device.launchApp({newInstance: true});
-    await device.sendToHome();
-    await device.launchApp({newInstance: false, url: url});
-    await expect(element(by.text(url))).toBeVisible();
+  [
+    {
+      platform: '',
+      ...withDefaultArgs(),
+    },
+    {
+      platform: 'android',
+      ...withAndroidSingleTaskActivityArgs(),
+    }
+  ].forEach(({platform, url, launchArgs}) => {
```

Comment:
I'm inclined to say that having explicit tests without looping, param preparation, and conditions is much more readable.
+ adding `android` to the platform name -> test name is a bit confusing in this way.

I suggest you duplicate the tests, and on Android only tests, such as signleInstance tests, add `:android:` to the test name. these tests will be ignored on iOS runs.

Reply:
I have to disagree in this case...